src/dft/backends/mklcpu/commit.cpp

```diff
@@ -31,109 +31,128 @@
-        DFT_ERROR status = DFT_NOTSET;
-        if (config_values.dimensions.size() == 1) {
-            status = DftiCreateDescriptor(&handle, get_precision(prec), get_domain(dom), 1,
-                                          config_values.dimensions[0]);
-        }
-        else {
-            status = DftiCreateDescriptor(&handle, get_precision(prec), get_domain(dom),
-                                          config_values.dimensions.size(),
-                                          config_values.dimensions.data());
-        }
-        if (status != DFTI_NO_ERROR) {
-            throw oneapi::mkl::exception("dft/backends/mklcpu", "commit",
-                                         "DftiCreateDescriptor failed");
-        }
-
-        set_value(handle, config_values);
+/// Commit impl class specialization for MKLCPU.
+template <dft::detail::precision prec, dft::detail::domain dom>
+class commit_derived_impl : public dft::detail::commit_impl {
+private:
+    static constexpr DFTI_CONFIG_VALUE mklcpu_prec = to_mklcpu(prec);
+    static constexpr DFTI_CONFIG_VALUE mklcpu_dom = to_mklcpu(dom);
+    using mklcpu_desc_t = DFTI_DESCRIPTOR_HANDLE;
 
-        status = DftiCommitDescriptor(handle);
+public:
+    commit_derived_impl(sycl::queue queue, const dft::detail::dft_values<prec, dom> config_values)
+            : dft::detail::commit_impl(queue, backend::mklcpu) {
+        DFT_ERROR status = DFTI_NOTSET;
+        sycl::buffer<DFT_ERROR, 1> status_buffer{ &status, sycl::range<1>{ 1 } };
+        sycl::buffer<mklcpu_desc_t, 1> handle_buffer{ &device_handle, sycl::range<1>{ 1 } };
+
+        queue.submit([&](sycl::handler& cgh) {
+            auto handle_obj =
+                handle_buffer.template get_access<sycl::access::mode::read_write>(cgh);
+            auto status_obj =
+                status_buffer.template get_access<sycl::access::mode::read_write>(cgh);
+
+            host_task<detail::kernel_name<mklcpu_desc_t>>(cgh, [=]() {
+                std::int64_t local_err = DFTI_MEMORY_ERROR;
+                if (config_values.dimensions.size() == 1)
+                    local_err = DftiCreateDescriptor(handle_obj.get_pointer(), mklcpu_prec,
+                                                     mklcpu_dom, 1, config_values.dimensions[0]);
+                else
+                    local_err = DftiCreateDescriptor(handle_obj.get_pointer(), mklcpu_prec,
+                                                     mklcpu_dom, config_values.dimensions.size(),
+                                                     config_values.dimensions.data());
+
+                set_value(*handle_obj.get_pointer(), config_values);
+                if (local_err != DFTI_NO_ERROR) {
+                    throw oneapi::mkl::exception("dft", "commit", "DftiCreateDescriptor failed");
+                }
+
+                local_err = DftiCommitDescriptor(*handle_obj.get_pointer());
+                *status_obj.get_pointer() = local_err;
+            });
+        });
```

Comment:
You resolved the previous comment so I'm adding a new one.
I don't believe anything is being gained by putting this in a host task, just added complexity.
This cannot be used for synchronization because there are no dependencies

Reply:
A descriptor can be recommited, more than once, so the descriptor should be synchronized

Reply:
@FMarno, this is changed post PR#282, not the constructor will create a descriptor and the commit with call DftiCommitDescriptor in a host_task, the descriptor `device_handle` will be wrapped in a buffer to manage syncronization as is expected for cases when recommit can happen.